src/tls13.c

```diff
@@ -850,6 +850,26 @@ static int DeriveTrafficSecret(WOLFSSL* ssl, byte* secret)
                      ssl->specs.mac_algorithm, 0);
 }
 
+
+static inline int Tls13_HKDF_Extract(WOLFSSL *ssl, byte* prk, const byte* salt, int saltLen,
+                                 byte* ikm, int ikmLen, int digest)
+{
+    int ret;
+#ifdef HAVE_PK_CALLBACKS
+        void *cb_ctx = ssl->HkdfExtractCtx;
+        CallbackHKDFExtract cb = ssl->ctx->HkdfExtractCb;
+        if ((cb != NULL) && (saltLen > 0)) {
+            ret = cb(prk, salt, saltLen, ikm, ikmLen, digest, cb_ctx);
+        } else {
+            ret = wc_Tls13_HKDF_Extract(prk, salt, saltLen, ikm, ikmLen, digest);
+        }
+#else
+        (void)ssl;
+        ret = wc_Tls13_HKDF_Extract(prk, salt, saltLen, ikm, ikmLen, digest);
+#endif
```

Comment:
I think it would be better to have the `HAVE_PK_CALLBACKS` guard only around the callback code. What I mean is something like:

```
        (void)ssl;
#ifdef HAVE_PK_CALLBACKS
        if (ssl->ctx->HkdfExtractCb != NULL) {
            /* callback call goes here */
        }
        else
#endif
        {
            ret = wc_Tls13_HKDF_Extract(prk, salt, saltLen, ikm, ikmLen, digest);
        }
```

This leaves a single `wc_Tls13_HKDF_Extract` which decreases the possibility of introducing bugs by modifying one call but not the other.

Reply:
Good idea. Fixed.